float32 maxP
float32 minP
---
bool ack